### PR TITLE
[Feature] Support RTMDet Ins Segmentation Inference

### PR DESCRIPTION
**Pull request by Nioolek**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation for this PR and the goal you want to achieve through this PR.

## Modification

Support RTMDet Ins Segmentation Inference

- [x] Inference

image
Can reproduce the map.

### CODE REVIEW

File: mmyolo/models/dense_heads/rtmdet_ins_head.py [hunk 1]
```diff
@@ -0,0 +1,649 @@
...
+        flatten_cls_scores = torch.cat(flatten_cls_scores, dim=1).sigmoid()
+        flatten_bbox_preds = torch.cat(flatten_bbox_preds, dim=1)
+        flatten_decoded_bboxes = self.bbox_coder.decode(
+            flatten_priors[..., :2].unsqueeze(0), flatten_bbox_preds,
+            flatten_stride)
+
+        flatten_kernel_preds = torch.cat(flatten_kernel_preds, dim=1)
+
+        # RTMDet does not have objectness prediction.
+        flatten_objectness = [None for _ in range(num_imgs)]
```
Comment (RangiLyu) [inline mmyolo/models/dense_heads/rtmdet_ins_head.py:413]: We can remove the objectness related code.

File: mmyolo/models/dense_heads/rtmdet_ins_head.py [hunk 2]
```diff
@@ -0,0 +1,743 @@
...
+                ori_h, ori_w = img_meta['ori_shape'][:2]
+                if isinstance(pad_param, np.ndarray):
+                    pad_param = pad_param.astype(np.int32)
+                    crop_y1, crop_y2 = pad_param[
+                        0], mask_logits.shape[-2] - pad_param[1]
+                    crop_x1, crop_x2 = pad_param[
+                        2], mask_logits.shape[-1] - pad_param[3]
+                    mask_logits = mask_logits[..., crop_y1:crop_y2,
+                                              crop_x1:crop_x2]
+                mask_logits = F.interpolate(
```
Comment (RangiLyu) [inline mmyolo/models/dense_heads/rtmdet_ins_head.py:646]: mmdet里面是先插值然后再crop，这里先crop再插值可能会有微小的差异

File: mmyolo/models/dense_heads/rtmdet_ins_head.py [hunk 3]
```diff
@@ -0,0 +1,722 @@
...
+            scale_factor = img_meta['scale_factor']
+            if 'pad_param' in img_meta:
+                pad_param = img_meta['pad_param']
+            else:
+                pad_param = None
+
+            score_thr = cfg.get('score_thr', -1)
+            if scores.shape[0] == 0:
+                empty_results = InstanceData()
+                empty_results.bboxes = bboxes
```
Comment (hhaAndroid) [inline mmyolo/models/dense_heads/rtmdet_ins_head.py:468]: 少了mask

File: mmyolo/models/dense_heads/rtmdet_ins_head.py (hunk 2 above)
Comment (Nioolek) [reply mmyolo/models/dense_heads/rtmdet_ins_head.py:628]: 由于预处理阶段使用了 `LetterResize` ，先resize图片，再上下左右pad。为了避免pad_param计算误差，所以这里是先crop，再resize。经过精度测试，s模型上该逻辑与mmdet中精度对齐，tiny模型上bbox map低0.1，mask map高0.1

Comment (RangiLyu) [reply mmyolo/models/dense_heads/rtmdet_ins_head.py:628]: OK

**Review by RangiLyu — APPROVED** [review]
